python/cog/types.py

```diff
@@ -47,12 +54,10 @@ def validate(cls, value: Any) -> io.IOBase:
         if isinstance(value, io.IOBase):
             return value
 
-        parsed_url = urlparse(value)
+        parsed_url = urllib.parse.urlparse(value)
         if parsed_url.scheme == "data":
-            # https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs
-            # TODO: decode properly. this maybe? https://github.com/fcurella/python-datauri/
-            header, encoded = parsed_url.path.split(",", 1)
-            return io.BytesIO(base64.b64decode(encoded))
+            res = urllib.request.urlopen(value)
+            return io.BytesIO(res.read())
```

Comment:
interesting change here - is this for efficiency/correctness?

Reply:
Correctness. The old code would work with base64-encoded data URIs but not arbitrary data URIs.

Reply:
I just read your initial PR comment - thx!